### DISCUSSION TITLE
Model backed attachments

### DISCUSSION THREAD

**Discussion by chrisgrande** [withlogicco/django-prose#99] [Ideas]
This is one I was [working on awhile ago](https://github.com/withlogicco/django-prose/compare/main...chrisgrande:django-prose:attachment-model), now a bit behind. 

One current issue this looks to solve is if you add and then remove attachments they will hang around in your backend, also attachments for deleted documents can't be easily purged. Moving to a model backed storage would be similar to how ActiveText handles file attachments with Trix.

**Comment by parisk** [comment]
Sounds like a great idea that could also fit in great with https://github.com/withlogicco/django-prose/issues/98.

Intuitively, I am thinking this could be a mixin to be used with any storage, plus a batteries included class for the default Django storage. How would you architect this?

**Reply by chrisgrande** [reply]
I was thinking an Attachments model, that could support files plus other types of possible embedded media. Possibly expandable so someone could easily add their own embeds. Sorry for the long delay in reply! Coming around to this, any thoughts on moving from Trix to [Lexxy](https://github.com/basecamp/lexxy)?

**Reply by chrisgrande** [reply]
I [did start this and have been hammering away a bit on it](https://github.com/chrisgrande/django-prose/pull/2), if there isn't any interest in moving to Lexxy in _django-prose_ I'll could fully fork it and make it's own package.
